### PR TITLE
feature/buying-a-reward

### PR DESCRIPTION
**Pull request by rniedziolka**
Hi,

This is a short cut show how this feature works:

https://www.loom.com/share/b31a5c3095da4644bbe3da8bad9514b7

### CODE REVIEW

File: app/models/employee.rb [hunk 1]
```diff
@@ -8,8 +8,11 @@ class Employee < ApplicationRecord
 
   has_many :given_kudos, class_name: 'Kudo', dependent: :destroy, inverse_of: :employee
   has_many :received_kudos, class_name: 'Kudo', foreign_key: 'receiver_id', dependent: :destroy, inverse_of: :receiver
+  has_many :orders, dependent: :nullify
+  has_many :rewards, through: :orders
 
   def likes
```
Comment (ciejj) [inline app/models/employee.rb]: the name of this method is misleading - I would consider something like `kudos_blance` or `kudos_score`

File: app/models/employee.rb [hunk 2]
```diff
@@ -8,8 +8,11 @@ class Employee < ApplicationRecord
...
     received_kudos.count
```
Comment (ciejj) [inline app/models/employee.rb]: I think this line is not needed here :)

File: db/seeds.rb [hunk 3]
```diff
@@ -22,5 +22,5 @@
 kudo3 = Kudo.create!(title: Faker::Food.dish, content: Faker::Food.description, employee_id: employee3.id, receiver_id: employee4.id, company_value: company_value1)
 kudo4 = Kudo.create!(title: Faker::Food.dish, content: Faker::Food.description, employee_id: employee4.id, receiver_id: employee3.id, company_value: company_value2)
 
-reward1 = Reward.create!(title: Faker::Food.dish, description: Faker::Food.description, price: 200)
-reward2 = Reward.create!(title: Faker::Food.dish, description: Faker::Food.description, price: 88)
+reward1 = Reward.create!(title: Faker::Food.dish, description: Faker::Food.description, price: 1)
```
Comment (ciejj) [inline db/seeds.rb:25]: if you are not using `reward1` anywhere else you could call `Reward.create!' without assigning it to the local variable

File: spec/system/reward_spec.rb [hunk 4]
```diff
@@ -29,5 +31,16 @@
     expect(page).to have_content reward.title
     expect(page).not_to have_content reward.description
     expect(page).to have_content reward.price
+
+    click_link 'Buy'
+    expect(page).to have_content 'You have insufficient funds in your account.'
+
+    create(:kudo, employee: employee, receiver: employee, company_value: company_value)
```
Comment (ciejj) [inline spec/system/reward_spec.rb]: Consider adding `company_value` to the kudos factory, so you won't have to specify it during `kudo` creation

**Review by ciejj — APPROVED** [review]
Good job - I have left some minor suggestions :)

**Comment by rniedziolka** [comment]
some fixes here:
b7cb11645c8e46c2a2051ef7b52befb988ed7565

**Pull request merged**